solution/ui/regulations/js/utils.js

```diff
@@ -233,6 +234,50 @@ const niceDate = (kebabDate) => {
     return `${month} ${day}, ${year}`;
 };
 
+/**
+ * Trap focus in an element (higher-order function)
+ * adapted from: https://hidde.blog/using-javascript-to-trap-focus-in-an-element/
+ *
+ * @param {HTMLElement} element - element in which to trap focus
+ *
+ * @returns {() => void} - function to remove event listener when invoked
+ */
+function trapFocus(element) {
+    const focusableEls = element.querySelectorAll(
+        'a[href]:not([disabled]), button:not([disabled]), textarea:not([disabled]), input[type="text"]:not([disabled]), input[type="radio"]:not([disabled]), input[type="checkbox"]:not([disabled]), select:not([disabled]), div:not([disabled]), iframe'
+    );
+
+    const firstFocusableEl = focusableEls[0];
+    const lastFocusableEl = focusableEls[focusableEls.length - 1];
+    const KEYCODE_TAB = 9;
+
+    const callbackFunc = (e) => {
+        const isTabPressed = e.key === "Tab" || e.keyCode === KEYCODE_TAB;
+
+        if (!isTabPressed) {
+            return;
+        }
+
+        if (e.shiftKey) {
+            /* shift + tab */ if (document.activeElement === firstFocusableEl) {
+                lastFocusableEl.focus();
+                e.preventDefault();
+            }
+        } /* tab */ else {
+            if (document.activeElement === lastFocusableEl) {
+                firstFocusableEl.focus();
+                e.preventDefault();
+            }
+        }
+    };
+
+    element.addEventListener("keydown", callbackFunc);
+
+    return function () {
+        element.removeEventListener("keydown", callbackFunc);
+    };
```

Comment:
returns a [closure](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Closures) that, when invoked, will remove the event listener that was created earlier in this method.